src/main/java/pl/datamatica/traccar/api/fcm/RegistrationConfirmResender.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (C) 2016  Datamatica (dev@datamatica.pl)
+ * 
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU Affero General Public License as published
+ *   by the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *  
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU Affero General Public License for more details.
+ *  
+ *   You should have received a copy of the GNU Affero General Public License
+ *   along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package pl.datamatica.traccar.api.fcm;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import pl.datamatica.traccar.api.providers.MailSender;
+import pl.datamatica.traccar.model.User;
+
+/**
+ *
+ * @author Jan Usarek
+ */
+public class RegistrationConfirmResender extends Thread {
+
+    private Thread t;
+    private final String threadName;
+    private final String emailTitle;
+    private final String emailMsg;
+    private final Logger logger;
+    private final String className;
+    private final MailSender sender;
+    private final User user;
+    private final static int NUM_OF_TRIES = 3;
+    private final static int SLEEP_TIME_SECONDS = 30;
+    
+    public RegistrationConfirmResender(MailSender sender, User user, String title, String msgContent) {
+        this.sender = sender;
+        this.user = user;
+        emailTitle = title;
+        emailMsg = msgContent;
+        className = RegistrationConfirmResender.class.getName();
+        threadName = className + "-" + user.getId();
```

Comment:
Nie ma ryzyka konfliktu? Co jeśli ktoś zażąda nowego maila zanim ta procedura się skończy i ten nowy też się nie uda?
Czy mogą być wątki o tej samej nazwie?

Reply:
Z dokumentacji Javy: "Every thread has a name for identification purposes. More than one thread may have the same name. If a name is not specified when a thread is created, a new name is generated for it."